arctic/chunkstore/chunkstore.py

```diff
@@ -0,0 +1,376 @@
+                                               SeriesSerializer,
+                                               PandasSerializer)
+from .date_chunker import DateChunker
+from ..exceptions import UnhandledDtypeException, NoDataFoundException
+from .._compression import compress_array, decompress
+
+
+logger = logging.getLogger(__name__)
+
+CHUNK_STORE_TYPE = 'ChunkStoreV1'
+
+
+class ChunkStore(object):
+    STRING_MAX = 16
+
+    @classmethod
+    def initialize_library(cls, arctic_lib, **kwargs):
+        ChunkStore(arctic_lib)._ensure_index()
+
+    @mongo_retry
+    def _ensure_index(self):
+        collection = self._collection.symbols
+        collection.create_index([("symbol", pymongo.ASCENDING),
+                                 ("_id", pymongo.DESCENDING)],
+                                background=True)
+
+        collection = self._collection
+        collection.create_index([('symbol', pymongo.HASHED)], background=True)
+        collection.create_index([('symbol', pymongo.ASCENDING),
+                                ('sha', pymongo.ASCENDING)],
+                                unique=True,
+                                background=True)
+        collection.create_index([('symbol', pymongo.ASCENDING),
+                                ('parent', pymongo.ASCENDING),
+                                ('start', pymongo.ASCENDING),
+                                ('end', pymongo.ASCENDING)],
+                                unique=True, background=True)
+
+    @mongo_retry
+    def __init__(self, arctic_lib, chunker=DateChunker()):
+        self.chunker = chunker
+        self._arctic_lib = arctic_lib
+
+        # Do we allow reading from secondaries
+        self._allow_secondary = self._arctic_lib.arctic._allow_secondary
+
+        # The default collection
+        self._collection = arctic_lib.get_top_level_collection()
+        self._symbols = self._collection.symbols
+
+    def __getstate__(self):
+        return {'arctic_lib': self._arctic_lib}
+
+    def __setstate__(self, state):
+        return ChunkStore.__init__(self, state['arctic_lib'])
+
+    def __str__(self):
+        return """<%s at %s>\n%s""" % (self.__class__.__name__, hex(id(self)),
+                                       indent(str(self._arctic_lib), 4))
+
+    def __repr__(self):
+        return str(self)
+
+    def delete(self, symbol):
+        """
+        Delete all chunks for a symbol.
+
+        Parameters
+        ----------
+        symbol : `str`
+            symbol name for the item
+        """
+        query = {"symbol": symbol}
+        self._collection.delete_many(query)
+        self._collection.symbols.delete_many(query)
+
+    def list_symbols(self):
+        """
+        Returns all symbols in the library
+
+        Returns
+        -------
+        list of str
+        """
+        return self._symbols.distinct("symbol")
+
+    def _get_symbol_info(self, symbol):
+        return self._symbols.find_one({'symbol': symbol})
```

Comment:
This method returns one (arbitrary) symbol inode. So you probably want to make sure:
`self._symbols.create_index(("symbol", pymongo.ASCENDING), unique=True)`